controller/hybridgateway/converter/http_route.go

```diff
@@ -216,16 +346,20 @@ func (c *httpRouteConverter) translate(ctx context.Context, logger logr.Logger)
 				WithControlPlaneRef(*cp).
 				WithOwner(c.route).Build()
 			if err != nil {
-				// TODO: decide how to handle build errors in converter
-				// For now, skip this resource
+				log.Error(logger, err, "Failed to build KongUpstream resource, skipping rule",
+					"upstream", upstreamName,
+					"controlPlane", cp.KonnectNamespacedRef)
+				translationErrors = append(translationErrors, fmt.Errorf("failed to build KongUpstream %s: %w", upstreamName, err))
 				continue
 			}
 			c.outputStore = append(c.outputStore, &upstream)
+			log.Debug(logger, "Successfully built KongUpstream resource",
+				"upstream", upstreamName)
 
 			// Build the KongTarget resources using the new rule-based approach.
 			targets, err := target.TargetsForBackendRefs(
 				ctx,
-				logr.Discard(), // TODO: pass proper logger.
+				logger.WithValues("upstream", upstreamName, "rule", utils.Hash32(rule.BackendRefs)),
```

Comment:
🤔  Why do we  put the Hash of BackendRefs in the logging? Isn't the upstream name enough?
```suggestion
				logger.WithValues("upstream", upstreamName),
```